lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +151,10 @@ def install
       end
     end
 
+    if @resource[:package_arch]
+      wanted = "#{wanted}.#{@resource[:package_arch]}"
+    end
+
```

Comment:
Hmm, possibly. The Yum provider uses the RPM uninstall code, might need some help plumbing that.